model/modelprocessor/index_doc_count_field.go

```diff
@@ -0,0 +1,61 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package modelprocessor
+
+import (
+	"context"
+	"sync"
+
+	"github.com/elastic/apm-server/model"
+	"github.com/elastic/beats/v7/libbeat/common"
+)
+
+// IndexDocCountField is a modelprocessor that sets whether or not
+// _doc_comment should be indexed on metricsets within a model.APMEvent.
+type IndexDocCountField struct {
+	mu sync.RWMutex
+	// ElasticSearch cluster version
+	version *common.Version
+}
+
+// SetESClusterVersion sets the elasticsearch cluster version, which is used to
+// decide whether _doc_count should be indexed or not.
+func (i *IndexDocCountField) SetESClusterVersion(version *common.Version) {
```

Comment:
I guess we shouldn't spend too much time on this, but I think it might be a bit neater if we had a sort of placeholder BatchProcessor which could be swapped with another one at runtime. Then, rather than updating the ES cluster version on this processor (which should only happen once), we would do something like this in beater:

```go
var esDocCountProcessorPlaceholder PlaceholderBatchProcessor
batchProcessor = append(batchProcessor, &esDocCountProcessorPlaceholder) // no-op until/unless set
...
g.Go(func() error {
	clusterInfo, err := s.waitReady(ctx, kibanaClient, indexDocCountField)
	if err != nil {
		return errors.Wrap(err, "error waiting for server to be ready")
	}
	if clusterInfo.Version != nil && clusterInfo.Version.LessThan(elasticsearchSupportsDocCount) {
		esDocCountProcessorPlaceholder.Set(modelprocessor.UnsetIndexDocCountField{})
	}
	return nil
})
```

Reply:
no worries, added something